node/cn/peer_test.go

```diff
@@ -169,3 +169,178 @@ func TestBasePeer_GetRW(t *testing.T) {
 	basePeer, pipe1, _ := newBasePeer()
 	assert.Equal(t, pipe1, basePeer.GetRW())
 }
+
+func TestBasePeer_SendBlockHeaders(t *testing.T) {
+	header1 := blocks[0].Header()
+	header2 := blocks[1].Header()
+
+	sentHeaders := []*types.Header{header1, header2}
+
+	basePeer, _, oppositePipe := newBasePeer()
+	go func(t *testing.T) {
+		if err := basePeer.SendBlockHeaders(sentHeaders); err != nil {
+			t.Fatal(err)
```

Comment:
Is calling `t.Fatal()` in this goroutine synced with the main goroutine?

Reply:
If `err != nil` is changed into `err == nil`, it keeps failing.